### DISCUSSION TITLE
Will requests with huge prameter imapct the memory usage of kong?

### DISCUSSION THREAD

**Discussion by yanansun2020** [Kong/kong#7981] [Help]
I'm recently debugging an OOM issue, and noticed when I have OOM, kong received a huge amount of requests, some of them are downloading files others are querying data with huge parameters. Wondering if that will affect kong memory usage?

Also, want to know what will cause kong memory usage except for the data added by admin API?

Thank you a lot!

**Comment by kikito** [comment]
Hello, files downloaded should not affect Kong, unless you are using some plugin which reads the whole response in memory (like proxy-cache or response-transformer). In general, it's always a good idea to report what plugins you are using when discussing problems with Kong.

Can you define a bit more what you mean with "huge parameters"? Are you talking about query parameters on the URL?

> what will cause kong memory usage except for the data added by admin API?

There are many things that can cause it, it's very difficult to list them all. Here are some possible things: too many basic entities (Routes, Services) too many Upstreams, too frequent config updates, badly written custom plugins, incorrectly configured nodes.

**Reply by yanansun2020** [reply]
@kikito thanks for the response. 
> Can you define a bit more what you mean with "huge parameters"? Are you talking about query parameters on the URL?

Yes

> it's always a good idea to report what plugins you are using when discussing

I can't put the exact plugins we use, but we only add about 150 jwt plugins, each plugin looks like below.
```
{
		"created_at": 1633128395092,
		"config": {
			"cookie": "***TOKEN",
			"claims_to_verify": ["exp"],
			"secret_is_base64": false,
			"key_claim_name": "key_name",
			"uri_param_names": ["jwt"]
		},
		"id": "9af71e4e-d9ab-4be2-92b0-5f68c9159c34",
		"enabled": true,
		"api_id": "21daafcf-fa54-4887-aa04-c0ecc063bd42",
		"name": "jwt"
	}
```
